plan/logical_plan_builder.go

```diff
@@ -409,6 +411,74 @@ func (b *planBuilder) buildSelection(p LogicalPlan, where ast.ExprNode, AggMappe
 	return selection
 }
 
+// buildProjectionFieldNameForColumns builds the field name and the table name when field expression is a column reference.
+func (b *planBuilder) buildProjectionFieldNameForColumns(field *ast.SelectField, c *expression.Column) (colName model.CIStr, tblName model.CIStr) {
+	if astCol, ok := getInnerFromParentheses(field.Expr).(*ast.ColumnNameExpr); ok {
+		colName = astCol.Name.Name
+		tblName = astCol.Name.Table
```

Comment:
return here `return astCol.Name.Name, astCol.Name.Table` and remove the `else` statement next line